src/Random.java

```diff
@@ -0,0 +1,192 @@
+        block[numStudents] = name;
+        numStudents++;
+    }
+
+    /*
+     * Randomization uses the studentsPerTable number given in App.Java to first
+     * randomize the names, then seperate them into groups
+     */
+    public String Randomization(int studentsPerTable) {
+        // all these variables have to be reset here because if randomization is
+        // recalled on (in case the teacher didn't like the previos random generated
+        // groups), their values would be at the end of the loop instead of the
+        // beginning
+        a = 0;
+        b = 1;
+        // b is set as 1 becuase it's used to label the groups as "Group 1, Group 2, etc
+              // "
+        i = 0;
+        randomizedGroups = "";
+        currentGroup = "";
+        leftoverGroup = numStudents % studentsPerTable;
+        /*
+         * This for loop is used to cycle through the names int eh array and randomize
+         * them. It works by setting the the variable nm as block[c], then giving
+         * block[c] the value of a random String in the block array, and giving where
+         * that String used to be the String of nm. I got this from Mr. Griswold's
+         * suggestion on how to randomize the groups
+         */
+        for (int c = 0; c < numStudents; c++) {
+            String nm = block[c];
+            int ran = (int) (Math.random() * numStudents);
+            block[c] = block[ran];
+            block[ran] = nm;
+        }
+        /*
+         * This next part seperates the now random Array into groups of
+         * StudentsPerTable. I cycles through numStudents, increasing both a and i,
+         * adding block[i] to a String called currentGroup. When a is equal to
+         * studentsPerTable, it adds currentGroup to the final Randomzation String,
+         * resets a and currentGroup, continuing until the loop is done
+         */
+        while (i < numStudents) {
+            currentGroup += block[i] + ", ";
+            a++;
+            if (a == studentsPerTable && i < numStudents - leftoverGroup) {
+                randomizedGroups += "Group " + b + ": " + currentGroup;
+                currentGroup = "";
+                a = 0;
+                b++;
+                
+            }
+            i++;
+        }
+        /*
+         * Because the textfile might not be evenly divided by studentsPerTable, this
+         * while loop accounts for the leftover groups that would've never been reached
+         * by (a == studentsPerTable)
+         */
+        while (leftoverGroup != 0 && i < numStudents) {
+            currentGroup += block[i] + ", ";
+            i++;
+        }
+        if (leftoverGroup != 0) {
+            randomizedGroups += "Group " + b + ": " + currentGroup;
+        }
+        /*
+         * This saves them to the textfile currentRandom, which changes everytime
+         * there's a new randomization
+         */
+        io.openWriteFile("currentRandom.txt");
```

Comment:
this writing should happen in the controller, App.java, not here. This should simply return the string and let the controller manage the IO work.